ts/components/DevicePairingDialog.tsx

```diff
@@ -0,0 +1,293 @@
+          >
+            {this.state.view === 'waitingForRequest' ? (
+              <div className="session-modal__centered">
+                <h3>{i18n('waitingForDeviceToRegister')}</h3>
+                <small className="text-subtle">
+                  {i18n('pairNewDevicePrompt')}
+                </small>
+                <div className="spacer-lg" />
+
+                <div id="qr">
+                  <QRCode value={window.textsecure.storage.user.getNumber()} />
+                </div>
+
+                <div className="spacer-lg" />
+                <div className="session-modal__button-group__center">
+                  <SessionButton
+                    text={i18n('cancel')}
+                    onClick={this.stopReceivingRequests}
+                  />
+                </div>
+              </div>
+            ) : (
+              <>
+                {this.state.data.length == 0 ? (
+                  <div className="session-modal__centered">
+                    <div>{i18n('noPairedDevices')}</div>
+                  </div>
+                ) : (
+                  <div className="session-modal__centered">
+                    {this.state.data.map((pubKey: any) => {
+                      const pubKeyInfo = this.getPubkeyName(pubKey);
+                      const isFinalItem =
+                        this.state.data[this.state.data.length - 1] === pubKey;
+
+                      return (
+                        <div key={pubKey}>
+                          <p>
+                            {pubKeyInfo.deviceAlias}
+                            <br />
+                            <span className="text-subtle">
+                              Pairing Secret:
+                            </span>{' '}
+                            {pubKeyInfo.secretWords}
+                          </p>
+                          {!isFinalItem ? (
+                            <hr className="text-soft fullwidth" />
+                          ) : null}
+                        </div>
+                      );
+                    })}
+                  </div>
+                )}
+
+                <div className="spacer-lg" />
+                <div className="session-modal__button-group__center">
+                  <SessionButton
+                    text={i18n('pairNewDevice')}
+                    onClick={this.startReceivingRequests}
+                  />
+                </div>
+              </>
+            )}
+          </SessionModal>
+        ) : null}
+      </>
+    );
+  }
+
+  private showView(
+    view?:
+      | 'default'
+      | 'waitingForRequest'
+      | 'requestReceived'
+      | 'requestAccepted'
+      | 'confirmUnpair'
+  ) {
+    if (!view) {
+      this.setState({
+        view: 'default',
+      });
+
+      return;
+    }
+
+    this.setState({ view });
+  }
+
+  private startReceivingRequests() {
+    this.setState({
+      isListening: true,
+    });
+
+    this.showView('waitingForRequest');
```

Comment:
both calls are updating the state. You could wrap them both in a single call to setState